### PR TITLE
clippy::perf fixes

### PR DESCRIPTION
**Pull request by matthiaskrgr**

### CODE REVIEW

File: src/concurrency_limiter.rs [hunk 1]
```diff
@@ -19,7 +19,7 @@ impl ConcurrencyLimiter {
         let available_token_condvar = Arc::new(Condvar::new());
 
         let state_helper = state.clone();
-        let available_token_condvar_helper = available_token_condvar.clone();
+        let available_token_condvar_helper = available_token_condvar;
```
Comment (bjorn3) [inline src/concurrency_limiter.rs:22]: Something when wrong when I introduced parallel compilation. `available_token_condvar` should have been moved into `ConcurrencyLimiter` after `available_token_condvar_helper` has been moved to the helper thread, but for some reason a new Condvar is stored in `ConcurrencyLimiter`. Fixed in 4a1c2d9c7c5cb370eea0df964d537b60d29a4191. Can you please rebase and revert this change?

Comment (matthiaskrgr) [reply src/concurrency_limiter.rs]: :+1:  done

**Comment by bjorn3** [comment]
Thanks!
